contrib/fparser/fparser_ad.cc

```diff
@@ -111,6 +114,95 @@ Value_t FunctionParserADBase<Value_t>::fp_plog(const Value_t * params)
   return x < a ? fp_log(a) + (x-a)/a - (x-a)*(x-a)/(Value_t(2)*a*a) + (x-a)*(x-a)*(x-a)/(Value_t(3)*a*a*a) : fp_log(x);
 }
 
+template <typename Value_t>
+Value_t FunctionParserADBase<Value_t>::fp_erf(const Value_t * params)
+{
+  const Value_t x = params[0];
+  return std::erf(x);
+}
+
+// Derivation of these complex method can be found here: https://math.stackexchange.com/a/712568/408963
+
+template<>
+std::complex<double>
+FunctionParserADBase<std::complex<double>>::fp_erf(const std::complex<double> * params)
+{
+  const std::complex<double> z = params[0];
+  double x = z.real(), y = z.imag();
+  double fx = std::erf(x) + std::exp(-(x*x)) / (2. * M_PI) * (1. - std::cos(2. * x * y));
+  double fy = std::exp(-(x*x)) / (2. * M_PI) * std::sin(2. * x * y);
+
+  // Set this to as high as you want for greater accuracy
+  unsigned n = 3;
+  for (unsigned k = 1; k <= n; k++)
+  {
+    double fxk = 2. * x * (1. - std::cos(2. * x * y) * std::cosh(k * y))
+      + k * std::sin(2. * x * y) * std::sinh(k * y);
+    double fyk = 2. * x * std::sin(2. * x * y) * std::cosh(k * y)
+      + k * std::cos(2. * x * y) * std::sinh(k * y);
+    double mult = 2. / M_PI * std::exp(-(x*x)) * std::exp(-k*k / 4.) / (k*k + 4. * x*x);
+    fx += mult * fxk;
+    fy += mult * fyk;
+  }
+
+  std::complex<double> f (fx, fy);
+  return f;
+}
+
+template<>
+std::complex<float>
+FunctionParserADBase<std::complex<float>>::fp_erf(const std::complex<float> * params)
+{
+  const std::complex<float> z = params[0];
+  float x = z.real(), y = z.imag();
+  double fx = std::erf(x) + std::exp(-(x*x)) / (2. * M_PI) * (1. - std::cos(2. * x * y));
+  double fy = std::exp(-(x*x)) / (2. * M_PI) * std::sin(2. * x * y);
+
+  // Set this to as high as you want for greater accuracy
+  unsigned n = 3;
+  for (unsigned k = 1; k <= n; k++)
+  {
+    double fxk = 2. * x * (1. - std::cos(2. * x * y) * std::cosh(k * y))
+      + k * std::sin(2. * x * y) * std::sinh(k * y);
+    double fyk = 2. * x * std::sin(2. * x * y) * std::cosh(k * y)
+      + k * std::cos(2. * x * y) * std::sinh(k * y);
+    double mult = 2. / M_PI * std::exp(-(x*x)) * std::exp(-k*k / 4.) / (k*k + 4. * x*x);
+    fx += mult * fxk;
+    fy += mult * fyk;
+  }
+
+  std::complex<float> f (fx, fy);
+  return f;
+}
+
+template<>
+std::complex<long double>
```

Comment:
This isn't really relevant now that we're using the Faddeeva stuff, but I'm pretty sure I figured out how I could create a single templated method for real calculations and another single templated method for complex calculations. The source would be like this I think:
```cpp
template<typename T>
template<typename U>
std::complex<U> FParserStuff<T>::fp_erf(const std::complex<U> x)
{
  /* Complex methods */
}

template<typename T>
template<typename U>
U FParserStuff<T>::fp_erf(const U x)
{
  /* Real methods */
}

template class FParserStuff<float>;
template class FParserStuff<std::complex<float>>;
template float FParserStuff<float>::fp_erf(float);
template std::complex<float> FParserStuff<std::complex<float>>::fp_erf(std::complex<float>);
```
I was concerned that the latter defined method would get called for `std::complex<float>` arguments but I checked in a debugger and the former method does get called (regardless of the ordering of the definitions). I guess the compiler chooses the more specific method?...